deepcell/image_generators/cropping.py

```diff
@@ -0,0 +1,576 @@
+                                    rescale=rescale,
+                                    preprocessing_function=preprocessing_function,
+                                    data_format=data_format,
+                                    validation_split=validation_split,
+                                    dtype=dtype)
+
+        if crop_size is not None:
+            if not isinstance(crop_size, (tuple, list)):
+                raise ValueError("Crop size must be a list or tuple of row/col dimensions")
+
+        self.crop_size = crop_size
+
+        # tensorflow does not initialize interpolation_order, so we'll do it here
+        self.interpolation_order = interpolation_order
+
+    def flow(self,
+             train_dict,
+             batch_size=1,
+             transforms=['outer-distance'],
+             transforms_kwargs={},
+             min_objects=3,
+             shuffle=True,
+             seed=None,
+             save_to_dir=None,
+             save_prefix='',
+             save_format='png'):
+        """Generates batches of augmented/normalized data with given arrays.
+
+        Args:
+            train_dict (dict): Consists of numpy arrays for X and y.
+            batch_size (int): Size of a batch. Defaults to 1.
+            shuffle (bool): Whether to shuffle the data between epochs.
+                Defaults to True
+            seed (int): Random seed for data shuffling.
+            min_objects (int): Minumum number of objects allowed per image
+            save_to_dir (str): Optional directory where to save the pictures
+                being yielded, in a viewable format. This is useful
+                for visualizing the random transformations being
+                applied, for debugging purposes.
+            save_prefix (str): Prefix to use for saving sample
+                images (if save_to_dir is set).
+            save_format (str): Format to use for saving sample images
+                (if save_to_dir is set).
+
+        Returns:
+            SemanticIterator: An Iterator yielding tuples of (x, y),
+                where x is a numpy array of image data and y is list of
+                numpy arrays of transformed masks of the same shape.
+        """
+        return CropperIterator(
+            train_dict,
+            self,
+            batch_size=batch_size,
+            transforms=transforms,
+            transforms_kwargs=transforms_kwargs,
+            shuffle=shuffle,
+            min_objects=min_objects,
+            seed=seed,
+            data_format=self.data_format,
+            save_to_dir=save_to_dir,
+            save_prefix=save_prefix,
+            save_format=save_format,
+            crop_size=self.crop_size)
+
+    def get_random_transform(self, img_shape, seed=None):
+        transform_parameters = ImageDataGenerator.get_random_transform(self, img_shape=img_shape,
+                                                                       seed=seed)
```

Comment:
Instead, you can use `super(CroppingDataGenerator, self).get_random_transform` (`CroppingDataGenerator` is a `SemanticGenerator` which is a `ImageDataGenerator`).

Reply:
Is that syntactically more clear than ImageDataGenerator.get_random_transform? 
Because then its obvious that its a superclass. The way its currently written there's no way to know its the parent class, right?

Reply:
Yes, it groups all the logic in a more clear way. Functionally, it's better because it keeps our functionality consistent.  If we went ahead and changed the `SemanticDataGenerator` in some way that overrides the `get_random_transform`, we want to be able to maintain those changes.

Using `super` just more naturally utilizes the object ineheritance.